src/index.js

```diff
@@ -1,49 +1,118 @@
-    const entries = Array.from($('article')).map(article =>
-      parseArticle($, article)
-    )
-    return addData(entries, 'io.cozy.books').then(() =>
-      saveFiles(entries, fields)
-    )
-  })
+async function start (fields) {
+  await login(fields)
+  const documents = await fetch()
+  await saveFiles(documents, fields)
 }
 
-// The goal of this function is to parse a html page wrapped by a cheerio instance // and return an array of js objects which will be saved to the cozy by addData (https://github.com/cozy/cozy-konnector-libs/blob/master/docs/api.md#module_addData)
-// and saveFiles (https://github.com/cozy/cozy-konnector-libs/blob/master/docs/api.md#savefiles)
-function parseArticle($, article) {
-  const $article = $(article)
-  const title = $article.find('h3 a').attr('title')
-  return {
-    title,
-    price: normalizePrice($article.find('.price_color').text()),
-    url: `${baseUrl}/${$article.find('h3 a').attr('href')}`,
-    // when it finds a fileurl attribute, saveFiles will save this file to the cozy with a filename
-    // name
-    fileurl: `${baseUrl}/${$article.find('img').attr('src')}`,
-    filename: `${title}.jpg`
+async function login (fields) {
+  log('info', 'Logging in')
+  await request(`${baseUrl}/LoginMDP`)
+  let $
+  try {
+    $ = await request({
+      method: 'POST',
+      uri: `${baseUrl}/LoginMDP?op=c&url=`,
+      form: {
+        url: '',
+        LMDP_Spi: fields.login,
+        LMDP_Password: fields.password,
+        LMDP_Spi_tmp: fields.login,
+        LMDP_Password_tmp: fields.password
+      }
+    })
+  } catch (err) {
+    log('error', 'Website failed while trying to login')
+    log('error', err.message)
+    throw new Error(errors.VENDOR_DOWN)
+  }
+
+  const erreurs = $('.erreur:not(.pasvisible)')
+  if (erreurs.length) {
+    log('error', erreurs.eq(0).text().trim())
+    throw new Error(errors.LOGIN_FAILED)
   }
 }
 
-// convert a price string to a float
-function normalizePrice(price) {
-  return parseFloat(price.trim().replace('£', ''))
+async function fetch () {
+  log('info', 'Fetching the list of documents')
+  let $ = await request(`${baseUrl}/acces-usager/cfs`)
+
+  // get the "Mes documents" link
+  const documentsLink = $('img[name=doc]').closest('a').attr('href')
+  const urlPrefix = documentsLink.split('/')[1] // gets "cesu-XX" from the url
+  $ = await request(`${baseUrl}${documentsLink}`)
+
+  const $form = $('form[name=documentsForm]')
+  const formUrl = $form.attr('action')
+  const token = $form.find('input[name=CSRFTOKEN]').val()
+
+  $ = await request({
+    method: 'POST',
+    uri: `${baseUrl}${formUrl}`,
+    form: {
+      annee: 'all',
+      CSRFTOKEN: token,
+      method: 'rechercheDocuments',
+      typeDocument: 'toutDocument',
+      typeImpot: 'toutImpot'
+    }
+  })
+
+  const documents = scrape($, {
+    fileurl: {
+      attr: 'onclick',
+      parse: onclick => {
+        const viewerUrl = onclick.match(/\((.*)\)/)[1].split(',')[0].slice(1, -1)
```

Comment:
I assume we're extracting args from a JS function call? I wonder about this on my own connectors... If we hardcode the function name inside the regexp, at least we don't end up with the arguments of another function in case another call is added in the event handler... But it would also mean renaming the function would break the connector (while your current solution would still work). I believe I would still prefer being explicit so we could exit with a better message in logs... WDYT?